### ISSUE TITLE
Support `let...else` syntax

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#10305]
- [ ] - Parsing `let...else`
- [ ] - Type inference 
- [ ] - Control flow analysis https://github.com/intellij-rust/intellij-rust/pull/10215 
- [ ] - Boolean operator (&& or ||) https://github.com/intellij-rust/intellij-rust/issues/10257
- [ ] - Improve parsing https://github.com/intellij-rust/intellij-rust/issues/10256
- [ ] - Improve diverging block in `let PATTERN: TYPE = EXPRESSION else DIVERGING_BLOCK;`

Tracking issue: https://github.com/rust-lang/rust/issues/87335
RFC: https://github.com/rust-lang/rfcs/pull/3137
[The Rust RFC Book](https://rust-lang.github.io/rfcs/3137-let-else.html#summary)

**Comment by AndreKR** [comment]
Yep, that was the issue, with new CLion and new plugin version the warning is gone.
